modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/ArchetyperArchetypeArtifactManager.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.project.templates.internal;
+
+import com.liferay.project.templates.internal.util.FileUtil;
+
+import java.io.File;
+import java.io.IOException;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+
+import java.util.Enumeration;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.apache.maven.archetype.common.DefaultArchetypeArtifactManager;
+import org.apache.maven.archetype.exception.UnknownArchetype;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+
+/**
+ * @author Gregory Amerson
+ */
+public class ArchetyperArchetypeArtifactManager
+	extends DefaultArchetypeArtifactManager {
+
+	public ArchetyperArchetypeArtifactManager(List<File> archetypesDirs)
```

Comment:
The variable `archetypesDirs` can hold directories or jar files so `archetypesFiles` seems like a better name.